extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -101,30 +106,45 @@ public static void write(
             @NotNull final Table t,
             @NotNull final TableDefinition definition,
             @NotNull final ParquetInstructions writeInstructions,
-            @NotNull final String destPathName,
+            @NotNull final File destFile,
+            @NotNull final File metadataFilePath,
             @NotNull final Map<String, String> incomingMeta,
-            final Map<String, GroupingColumnWritingInfo> groupingColumnsWritingInfoMap)
+            final Map<String, GroupingColumnWritingInfo> groupingColumnsWritingInfoMap,
+            @NotNull final MetadataFileWriterBase metadataFileWriter)
             throws SchemaMappingException, IOException {
         final TableInfo.Builder tableInfoBuilder = TableInfo.builder();
         List<File> cleanupFiles = null;
         try {
             if (groupingColumnsWritingInfoMap != null) {
                 cleanupFiles = new ArrayList<>(groupingColumnsWritingInfoMap.size());
-                final Path destDirPath = Paths.get(destPathName).getParent();
-                for (Map.Entry<String, GroupingColumnWritingInfo> entry : groupingColumnsWritingInfoMap.entrySet()) {
+                final Path destDirPath = destFile.toPath().getParent();
+                for (final Map.Entry<String, GroupingColumnWritingInfo> entry : groupingColumnsWritingInfoMap
+                        .entrySet()) {
                     final String groupingColumnName = entry.getKey();
                     final Table auxiliaryTable = groupingAsTable(t, groupingColumnName);
                     final String parquetColumnName = entry.getValue().parquetColumnName;
-                    final File metadataFilePath = entry.getValue().metadataFilePath;
+                    final File metadataGroupingFilePath = entry.getValue().metadataFilePath;
                     final File groupingDestFile = entry.getValue().destFile;
                     cleanupFiles.add(groupingDestFile);
                     tableInfoBuilder.addGroupingColumns(GroupingColumnInfo.of(parquetColumnName,
-                            destDirPath.relativize(metadataFilePath.toPath()).toString()));
-                    write(auxiliaryTable, auxiliaryTable.getDefinition(), writeInstructions,
-                            groupingDestFile.getAbsolutePath(), Collections.emptyMap(), TableInfo.builder());
+                            destDirPath.relativize(metadataGroupingFilePath.toPath()).toString()));
+                    // We don't include schema and row group information from grouping files in the metadata files
+
+                    // TODO Remove this comment
```

Comment:
Remove TODO when done

Reply:
We need to merge this "manually". Use `io.deephaven.parquet.table.metadata.TableInfo#deserializeFromJSON` to turn values into `TableInfo`, and `io.deephaven.parquet.table.metadata.TableInfo#serializeToJSON` to make new values.

`groupingColumns`, `dataIndexes` should probably always be dropped.
`sortingColumns` and `version` are optional.
`columnTypes` must be the same for all partitions, I think.

Reply:
I added this logic but after this, adding metadata from grouping files to the overall metadata files has a weird behavior. Consider the following screenshot:

![Screenshot 2024-02-16 at 11 11 13 PM](https://github.com/deephaven/deephaven-core/assets/16147908/4b873f4a-731d-4432-a395-e509871cf286)

We can see that the combined schema now has columns dh_key, dh_begin_pos and dh_end_pos and the blocks has references to the grouping file (block[0] has path ".dh_metadata/indexes/vvv/index_vvv_parquetWithGroupingDataAndMetadataTest.parquet"). So now if I do a read on the directory, it would go to the metadata file, locate all these blocks (including the ones from the grouping file) and add all these columns to the main table. I can walk you through this on a screen share session as well.